lib/collection/src/shards/replica_set.rs

```diff
@@ -738,6 +738,22 @@ impl ShardReplicaSet {
         res && !self.is_locally_disabled(peer_id)
     }
 
+    /// Determines if the given peer is local and is in `Initializing` state.
+    pub async fn peer_is_local_initializing(&self, peer_id: &PeerId) -> bool {
```

Comment:
Can we just make this function about checking `peer_is_initializing`?

It would make it less specific and then it could be shortcut by 

```rust
if &self.this_peer_id() == peer_id && self.is_local().await && self.is_initializing()`
```

Reply:
Good suggestion. I'll change this. Thanks!

Reply:
Implemented in 99103674934092431d3342503d20842764e16250